nixos/modules/tasks/network-interfaces.nix

```diff
@@ -541,6 +544,28 @@ in
       '';
     };
 
+    networking.machineId = mkOption {
```

Comment:
This should be a file option. To quote the docs:

> This ID uniquely identifies the host. It should be considered "confidential", and must not be exposed in untrusted environments, in particular on the network. If a stable unique identifier that is tied to the machine is needed for some application, the machine ID or any part of it must not be used directly. Instead the machine ID should be hashed with a cryptographic, keyed hash function, using a fixed, application-specific key. That way the ID will be properly unique, and derived in a constant way from the machine ID but there will be no way to retrieve the original machine ID from the application-specific one.

https://www.freedesktop.org/software/systemd/man/latest/machine-id.html

Reply:
The machine ID should be readable to all applications running on the local machine. If you don't expose the Nix store over the network or publish your configuration (e.g. on GitHub) it's fine. However for those two use-cases it would indeed be useful to provide the machine ID as a file outside of the Nix store. However then we lose the ability to automatically derive the hostid from the machine ID.
The docs you mention bring up another very important point: We must not derive the hostid by just taking a slice of the machine ID and need to instead use the procedure from the docs. I actually have this implemented properly in my local config and will provide a fix later.

Reply:
I should be fine to it as a option, and not as a file. The default permissions for `/etc/machine-id` is `-r--r--r-- 1 root root`, so it's world readable as @Luflosi also pointed out.

Fixing that the hostid is directly derived from machineid is fixed in the next  push (hopefully).